projects/packages/forms/src/blocks/form-step-navigation/edit.js

```diff
@@ -0,0 +1,206 @@
+
+	// Track the current step index (used later when replacing inner blocks)
+	let currentIndex = 0;
+
+	if ( isOutsideSteps && isSingleStep && activeStepId ) {
+		// When outside steps but in single step mode, show buttons based on the active step
+		const activeStepIndex = steps.findIndex( block => block.clientId === activeStepId );
+		if ( activeStepIndex !== -1 ) {
+			currentIndex = activeStepIndex;
+		}
+	} else if ( ! isOutsideSteps ) {
+		// Inside a step - determine position
+		const stepIndex = steps.findIndex( block => block.clientId === ancestorStepClientId );
+		currentIndex = stepIndex;
+	}
+
+	const { navigationBlocks } = useSelect(
+		select => {
+			const { getBlocks } = select( blockEditorStore );
+			const currentNavigationBlocks = getBlocks( clientId );
+			return {
+				navigationBlocks: currentNavigationBlocks,
+			};
+		},
+		[ clientId ]
+	);
+
+	const innerBlocksProps = useInnerBlocksProps( blockProps, {
+		template: NAVIGATION_TEMPLATE,
+		templateLock: 'all',
+		allowedBlocks: [ 'jetpack/button' ],
+		renderAppender: false,
+	} );
+
+	useEffect( () => {
+		// This happends when the step block is duplilcated
+		if ( typeof currentIndex === 'undefined' ) {
+			return undefined;
+		}
+		let shouldReplaceInnerBlocks = false;
+
+		// First identify existing buttons in the navigation
+		const existingButtons = {
+			previous: navigationBlocks.find(
+				block => block.name === 'jetpack/button' && block.attributes.uniqueId === 'previous-step'
+			),
+			next: navigationBlocks.find(
+				block => block.name === 'jetpack/button' && block.attributes.uniqueId === 'next-step'
+			),
+			submit: navigationBlocks.find(
+				block => block.name === 'jetpack/button' && block.attributes.uniqueId === 'submit-step'
+			),
+		};
+
+		// Create a map of button types to track required changes
+		const buttonUpdates = {
+			'previous-step': { needed: false, existing: existingButtons.previous },
+			'next-step': { needed: false, existing: existingButtons.next },
+			'submit-step': { needed: false, existing: existingButtons.submit },
+		};
+
+		// Flag needed buttons based on template
+		NAVIGATION_TEMPLATE.forEach( ( [ , blockAttributes ] ) => {
+			buttonUpdates[ blockAttributes.uniqueId ].needed = true;
+
+			// If button doesn't exist but is needed, we'll need to replace inner blocks
+			if ( ! buttonUpdates[ blockAttributes.uniqueId ].existing ) {
+				shouldReplaceInnerBlocks = true;
+			}
+		} );
+
+		// Build the updated button collection
+		const replacementInnerBlocks = NAVIGATION_TEMPLATE.map( ( [ blockName, blockAttributes ] ) => {
+			return (
+				buttonUpdates[ blockAttributes.uniqueId ].existing ||
+				createBlock( blockName, {
+					...blockAttributes,
+					uniqueId: blockAttributes.uniqueId,
+				} )
+			);
+		} );
+
+		if ( shouldReplaceInnerBlocks ) {
+			__unstableMarkNextChangeAsNotPersistent();
+			replaceInnerBlocks( clientId, replacementInnerBlocks, false );
+			return undefined;
```

Comment:
Nit, I think the `undefined` isn't needed:
```suggestion
			return;
```

Reply:
Thanks! this should be fixed.